src/lib.rs

```diff
@@ -81,6 +83,8 @@ where
         Lowercase,
         /// The previous cased character in the current word is uppercase.
         Uppercase,
+        /// The previous cased character in the current word is numeric
+        Numeric,
```

Comment:
I would have expected the logic change in transform to look different, but I also haven't thought about it much. Will review this part in more detail later.